### PR TITLE
Allow numeric defaults for booleans

### PR DESCRIPTION
**Pull request by abhinav**
Also, fix parser bug that prevented starting variable names with true or false

@blampe @breerly @junchaowu

### CODE REVIEW

**Comment by junchaowu** [comment]
lgtm

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,7 +4,10 @@ Releases
 0.5.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Fix a bug in the parser that prevented starting identifier names with
+  ``true`` or ``false``.
+- Allow passing 0 and 1 as default values for ``bool``. These will
+  automatically be casted to boolean.
```
Comment (blampe) [inline CHANGELOG.rst]: casted? cast? Need a grammar nerd up in here.

**Comment by blampe** [comment]
:+1:

File: CHANGELOG.rst (hunk 1 above)
Comment (abhinav) [inline CHANGELOG.rst]: You might be right. Switching to cast.

**Pull request merged**